ansible/playbooks/rate_limit_setup.yml

```diff
@@ -0,0 +1,79 @@
+- name: Istio Rate Limiting Setup
+  hosts: all
+  become: true
+  vars:
+    service_name: model-service 
+    namespace: default
+
+  tasks:
+    - name: Download rate limit service manifest
+      ansible.builtin.get_url:
+        url: https://raw.githubusercontent.com/istio/istio/release-1.20/samples/ratelimit/rate-limit-service.yaml
+        dest: /tmp/rate-limit-service.yaml
+
+    - name: Apply rate limit service
+      kubernetes.core.k8s:
+        state: present
+        src: /tmp/rate-limit-service.yaml
+        namespace: istio-system
+      environment:
+        K8S_AUTH_KUBECONFIG: /home/vagrant/.kube/config
+
+
+    - name: Create EnvoyFilter for rate limiting
+      kubernetes.core.k8s:
+        state: present
+        definition:
+          apiVersion: networking.istio.io/v1alpha3
+          kind: EnvoyFilter
+          metadata:
+            name: rate-limit
+            namespace: istio-system
+          spec:
+            workloadSelector:
+              labels:
+                istio: ingressgateway
+            configPatches:
+              - applyTo: HTTP_FILTER
+                match:
+                  context: GATEWAY
+                  listener:
+                    portNumber: 80
+                    filterChain:
+                      filter:
+                        name: envoy.filters.network.http_connection_manager
+                        subFilter:
+                          name: envoy.filters.http.router
+                patch:
+                  operation: INSERT_BEFORE
+                  value:
+                    name: envoy.filters.http.ratelimit
+                    typed_config:
+                      "@type": type.googleapis.com/envoy.extensions.filters.http.ratelimit.v3.RateLimit
+                      domain: echo-ratelimit
+                      failure_mode_deny: true
+              - applyTo: CLUSTER
+                match:
+                  context: ANY
+                patch:
+                  operation: ADD
+                  value:
+                    name: rate_limit_cluster
+                    type: STRICT_DNS
+                    connect_timeout: 0.25s
+                    lb_policy: ROUND_ROBIN
```

Comment:
Are you using round robin intentionally? There's also the ring hash option - idk if you've looked into it; it gives us consistent LB resolution out of the box.